python/feathub/processors/spark/spark_table.py

```diff
@@ -0,0 +1,120 @@
+# limitations under the License.
+import typing
+from collections import defaultdict
+from datetime import timedelta
+from typing import Optional
+
+import pandas as pd
+from pyspark.sql import DataFrame as NativeSparkDataFrame
+
+from feathub.common.exceptions import FeathubException
+from feathub.common.types import to_numpy_dtype
+from feathub.feature_tables.feature_table import FeatureTable
+from feathub.feature_tables.sinks.file_system_sink import FileSystemSink
+from feathub.feature_tables.sinks.print_sink import PrintSink
+from feathub.processors.spark.spark_job import SparkJob
+from feathub.processors.spark.spark_types_utils import (
+    to_feathub_schema,
+    to_feathub_type,
+)
+from feathub.table.schema import Schema
+from feathub.table.table import Table
+from feathub.table.table_descriptor import TableDescriptor
+
+if typing.TYPE_CHECKING:
+    from feathub.processors.spark.spark_processor import SparkProcessor
+
+
+class SparkTable(Table):
+    """
+    The implementation of Feathub Table for Spark.
+
+    It can only be instantiated and processed by SparkProcessor. It converts the feature
+    to native Spark DataFrame internally with the SparkDataFrameBuilder.
+    """
+
+    def __init__(
+        self,
+        spark_processor: "SparkProcessor",
+        feature: TableDescriptor,
+    ) -> None:
+        """
+        Instantiate the SparkTable.
+
+        :param spark_processor: The SparkProcessor that instantiate this SparkTable.
+        :param feature: Describes the features to be included in the table. If it is a
+                        FeatureView, it must be resolved.
+        """
+        super().__init__(feature.timestamp_field, feature.timestamp_format)
+        self.spark_processor = spark_processor
+        self.feature = feature
+
+    def get_schema(self) -> Schema:
+        return to_feathub_schema(self._get_spark_dataframe(self.feature).schema)
+
+    def to_pandas(self, force_bounded: bool = False) -> pd.DataFrame:
+        feature = self.feature
+        if not feature.is_bounded():
+            if not force_bounded:
+                raise FeathubException(
+                    "Unbounded table cannot be converted to Pandas DataFrame. You can "
+                    "set force_bounded to True to convert the Table to DataFrame."
+                )
+            feature = feature.get_bounded_view()
+
+        dataframe = self._get_spark_dataframe(feature)
+
+        field_names = []
+        field_types = []
+        for field in dataframe.schema.fields:
+            field_names.append(field.name)
+            field_types.append(to_numpy_dtype(to_feathub_type(field.dataType)))
+
+        results = dataframe.collect()
+
+        data: typing.Dict[str, typing.List[typing.Any]] = defaultdict(list)
+        for row in results:
+            for name, value in zip(field_names, row):
+                data[name].append(value)
+
+        values = [value for _, value in data.items()]
+        return pd.DataFrame(
```

Comment:
Would it be simpler to use `DataFrame.toPandas()` to get Pandas dataframe from Spark?